src/Arcus.WebApi.Security/Authentication/CertificateAuthenticationFilter.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using GuardNet;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Arcus.WebApi.Security.Authentication
+{
+    /// <summary>
+    /// Authentication filter to secure HTTP requests by allowing only certain values in the client certificate.
+    /// </summary>
+    public class CertificateAuthenticationFilter : IAuthorizationFilter
+    {
+        private readonly X509ValidationRequirement _requirement;
+        private readonly string _expectedValue;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CertificateAuthenticationFilter"/> class.
+        /// </summary>
+        /// <param name="requirement">The property of the client <see cref="X509Certificate2"/> to validate.</param>
+        /// <param name="expectedValue">The expected value the property of the <see cref="X509Certificate2"/> should have.</param>
+        public CertificateAuthenticationFilter(X509ValidationRequirement requirement, string expectedValue)
+        {
+            Guard.NotNullOrWhitespace(expectedValue, nameof(expectedValue), "Expected value in certificate cannot be blank");
+
+            _requirement = requirement;
+            _expectedValue = expectedValue;
```

Comment:
I can see value for this, but we should use be more flexible than this. We don't want to restart because the value has changed.

So I would suggest providing a way where we can specify how to retrieve it via `IConfiguration`

Reply:
Wouldn't it be an idea to create a feature-request for this, and implement that in a separate pull-request ?

Reply:
Personally, I don't see it as a feature request given you will never add an attribute with a fixed thumbprint for example. It will always come from the configuration, no?

Does that mean we should not have that option? I don't think so.

But I get your point of doing seperate PR to simplify this one, tracking in #44